test/e2e/performanceprofile/functests/11_mixedcpus/mixedcpus.go

```diff
@@ -131,7 +131,7 @@ func setup(ctx context.Context, profile *performancev2.PerformanceProfile) func(
 	mcps.WaitForCondition(mcp, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
 	mcps.WaitForCondition(mcp, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
 
-	teardown := func(ctx context.Context) {
+	teardown := func() {
```

Comment:
why  we want to remove the `ctx` argument here?

Reply:
The `teardown` function is a value returned by the `setup` function and will be using the context that was passed initially to `setup` instead of accepting new one.

Reply:
this works, but it means that the same context (so possibly the same deadline) will cover setup + actual run (which spans between setup and teardown) + teardown. Previously we had a context for setup, a different context for teardown, and the actual run was not guarded by a context. Reducing to just one seems too much of a stretch.

Reply:
Ack. Fixed that, thank you for your input